extensions/wikia/AdEngine/js/template/bfaaMobile.js

```diff
@@ -0,0 +1,100 @@
+/*global define, require*/
+define('ext.wikia.adEngine.template.bfaaMobile', [
+	'ext.wikia.adEngine.adHelper',
+	'ext.wikia.adEngine.context.uapContext',
+	'ext.wikia.adEngine.provider.btfBlocker',
+	'ext.wikia.adEngine.slotTweaker',
+	'ext.wikia.adEngine.video.uapVideoAd',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window',
+	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
+], function (
+	adHelper,
+	uapContext,
+	btfBlocker,
+	slotTweaker,
+	uapVideoAd,
+	doc,
+	log,
+	win,
+	mercuryListener
+) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.template.bfaaMobile',
+		page,
+		unblockedSlots = [
+			'MOBILE_BOTTOM_LEADERBOARD',
+			'MOBILE_IN_CONTENT',
+			'MOBILE_PREFOOTER'
+		],
+		wrapper;
+
+	function runOnReady(iframe, params) {
+		var aspectRatio = params.aspectRatio,
+			adsModule = win.Mercury.Modules.Ads.getInstance(),
+			adjustPadding = function () {
+				var viewPortWidth = Math.max(doc.documentElement.clientWidth, win.innerWidth || 0),
+					height = aspectRatio ?
+						viewPortWidth / aspectRatio : iframe.contentWindow.document.body.offsetHeight;
+				page.style.paddingTop = height + 'px';
+				adsModule.setSiteHeadOffset(height);
+			},
+			onResize = adHelper.throttle(adjustPadding, 100);
+
+		page.classList.add('bfaa-template');
+		adjustPadding();
+
+		win.addEventListener('resize', onResize);
+		if (mercuryListener) {
+			mercuryListener.onPageChange(function () {
+				page.classList.remove('bfaa-template');
+				page.style.paddingTop = '';
+				adsModule.setSiteHeadOffset(0);
+				win.removeEventListener('resize', onResize);
+			});
+		}
+
+		if (params.videoUrl && params.videoTriggerElement) {
+			slotTweaker.onReady(params.slotName, function(iframe) {
+				var imageContainer = iframe.parentNode.parentNode.parentNode,
```

Comment:
• get top leaderboard by slot name and then get first div child
• walk up from iframe till parenNode equals top leaderboard
What do you prefer (any other ideas)?